global/scripts/parallel.shlib

```diff
@@ -38,6 +38,12 @@ function par_addjob()
 
 function par_runjobs()
 {
+    #check if there are any jobs added
+    if (( ${#par_jobarray[@]} == 0 )); then 
+        echo "WARNING: no parallel jobs added, nothing to run"
+        return $? #return code of the prevously run command - for compatibility with NONE mode
```

Comment:
I don't like this, the return code of something in parallel.shlib shouldn't try to depend on whatever random thing was run before it.  Furthermore, this `$?` won't do what is intended, it will return the exit code of the `echo` line.
```suggestion
        return 0
```

Reply:
In other words, just as the warning you added implies, you shouldn't even try to run this function if you didn't add any jobs.

Reply:
This is not used in the latest design, removed.